src/com/biocollapse/service/InfectionService.java

```diff
@@ -1,5 +1,146 @@
+        int infectionProbability = config.getInfectionProbability();
+        int effectiveInfectionProbability = config.getMaskMandate() ? infectionProbability / 2 : infectionProbability;
+
+        // Check surrounding positions within the infection radius
+        for (MapPosition pos : infectedPos){
+            for (MapPosition neighbor : getPositionsWithinRadius(pos, infectionRadius)) {
+                Human neighborHuman = findHumanAtPosition(neighbor, humans);
+                if (neighborHuman != null && neighborHuman.isAlive() && !neighborHuman.isInfected() && !neighborHuman.isImmune()) {
+                    if (GlobalRandom.checkProbability(effectiveInfectionProbability)) {
+                        neighborHuman.setInfected(true);
+                        neighborHuman.setInfectedAt(currentTick);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * calculate the positions located in a specific radius
+     * @param pos
+     * @param radius
+     * @return a list of MapPositions in a specific radius
+     */
+    private List<MapPosition> getPositionsWithinRadius(MapPosition pos, int radius) {
+        List<MapPosition> neighbors = new ArrayList<>();
+
+        // Loop through the square grid around the human, bounded by the infection radius
+        for (int i = -radius; i <= radius; i++) {
+            for (int j = -radius; j <= radius; j++) {
+                MapPosition neighborPos = new MapPosition(pos.getRow() + i, pos.getCol() + j);
+                if (isWithinMapBounds(neighborPos)) {
+                    neighbors.add(neighborPos);
+                }
+            }
+        }
+
+        return neighbors;
+    }
+
+    /**
+     * checks of a calculated position is within the bounds of the Map
+     * @param pos
+     * @return boolean
+     */
+    private boolean isWithinMapBounds(MapPosition pos) {
+        return pos.getRow() >= 0 && pos.getRow() < Map.MAP_HEIGHT && pos.getCol() >= 0 && pos.getCol() < Map.MAP_WIDTH;
+    }
+
+    /**
+     * finds the human located in  specific MapPosition 
+     * @param position
+     * @param humans
+     * @return Human
+     */
+    private Human findHumanAtPosition(MapPosition position, Human[] humans) {
+        for (Human human : humans) {
+            if (human.getPos().equals(position)) {
+                return human;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * updates the human list with new healed of dead humans according to the parameters
+     * @param humans
+     * @param config
+     * @param currentTick
+     */
+    private void updateDeadOrHealed(Human[] humans, Config config, int currentTick) {
+        int mortalityRisk = config.getMortalityRisk();
+
+        for (Human human : humans) {
+            if (human.isInfected()) {
+                //hospitalization decreases mortality risk 
+                int effectiveMortalityRisk = human.isHospitalized() ? mortalityRisk / 4 : mortalityRisk;
```

Comment:
It's a great idea to reduce the probability of dying as soon as someone is in hospital. But I would also add a new parameter in the config as above. So that we have an overview of all such values there.

Reply:
We can definitly check that later as an improvement to the code.

Reply:
Sounds good to me 👍🏻